python/cuvs/cuvs/common/exceptions.pyx

```diff
@@ -28,7 +28,7 @@ def get_last_error_text():
     if c_err is NULL:
         return
     cdef bytes err = c_err
-    return err.decode("utf8")
+    return err.decode("utf8", "ignore")
```

Comment:
thats a great question!

So I originally had a bug in the ivf_pq deserialization code, that resulted in some bad data being passed to the  mdspan serializer, which caused an error to be thrown that wasn't valid utf8 text.

Specifically this line https://github.com/rapidsai/raft/blob/b66b269ab6dcda48aef3a6ed9e7f604e99471d72/cpp/include/raft/core/detail/mdspan_numpy_serializer.hpp#L293  was writing out the error message `unrecognized byteorder %c` where the `%c` was pointing to some random data (`\x93` in my case) - which couldn't get converted to utf8. This led to a unicode decode error being thrown rather than the actual error.

While I think we might want to consider always having error messages that are valid utf8 text (like converting that `%c` format to something like `%x` ) - I don't think we can guarantee that the exception text from our c++ layer is valid utf8, and ignoring any conversion errors results in a better error message than failing on the utf8 decode.

Reply:
That makes a lot of sense, and it's great to know! Thanks for the detailed explanation :)